scanomatic/generics/abstract_model_factory.py

```diff
@@ -144,9 +144,9 @@ class AbstractModelFactory:
     MODEL = Model
 
     _LOGGER = None
-    _SUB_FACTORIES = dict()
+    _SUB_FACTORIES: Dict[Model, "AbstractModelFactory"] = dict()
     STORE_SECTION_HEAD = ""
-    STORE_SECTION_SERIALIZERS = dict()
+    STORE_SECTION_SERIALIZERS: Dict[str, Any] = dict()
```

Comment:
when adding or updating types prefer the lower case `dict[str, Any]`  instead of importing `Dict`, `Tuple` or `List`. No need to change all usage of it but we should convert to these as they don't require imports as we go along. Not sure what version of python this was introduced, but it works for 3.9 at least.

Reply:
This is also interesting https://www.python.org/dev/peps/pep-0589/

Reply:
I'm pretty sure it was introduced in 3.9.

Reply:
As I write below, I'd actually prefer to take care of all of these at once in a separate PR, rather than doing it piecemeal.

Reply:
that's fine if you think it's a simple thing to fix them all, then better do them separately. If not then please fix as you go along. Your choice.